src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -1408,6 +1408,7 @@ TwoPhaseInsertSelectTaskList(Oid targetRelationId, Query *insertSelectQuery,
 		 * Generate a query string for the query that inserts into a shard and reads
 		 * from an intermediate result.
 		 */
+		insertResultQuery->cteList = NIL;
```

Comment:
maybe add a comment on why we need to do this

Reply:
Done